ship/src/test/scala/ch/epfl/bluebrain/nexus/ship/resources/DistributionPatcherSuite.scala

```diff
@@ -0,0 +1,109 @@
+package ch.epfl.bluebrain.nexus.ship.resources
+
+import cats.effect.IO
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.FileSelf
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.FileSelf.ParsingError.InvalidFileId
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.nxv
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{BaseUri, ResourceUris}
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.{Label, ProjectRef, ResourceRef}
+import ch.epfl.bluebrain.nexus.ship.ProjectMapper
+import ch.epfl.bluebrain.nexus.testkit.mu.NexusSuite
+
+class DistributionPatcherSuite extends NexusSuite {
+
+  private val project1       = ProjectRef.unsafe("bbp", "proj1")
+  private val targetProject1 = ProjectRef.unsafe("obp", "proj1")
+
+  private val resourceIri: Iri = nxv + "resourceId"
+
+  private val originalBase = BaseUri(uri"http://bbp.epfl.ch/nexus", Label.unsafe("v1"))
+  private val targetBase   = BaseUri(uri"https://www.openbrainplatform.org/api/nexus", Label.unsafe("v1"))
+
+  private val validFileSelfUri = buildFileSelfUri(project1, resourceIri).accessUri(originalBase)
+
+  private def buildFileSelfUri(project: ProjectRef, id: Iri) =
+    ResourceUris("files", project, id)
+
+  private val fileSelf = new FileSelf {
+    override def parse(input: IriOrBNode.Iri): IO[(ProjectRef, ResourceRef)] =
+      input match {
+        case value if value.startsWith(originalBase.iriEndpoint) => IO.pure(project1 -> ResourceRef.Latest(resourceIri))
+        case value                                               => IO.raiseError(InvalidFileId(value))
+      }
+  }
+
+  private val patcherNoProjectMapping   = new DistributionPatcher(fileSelf, ProjectMapper(Map.empty), targetBase)
+  private val patcherWithProjectMapping =
+    new DistributionPatcher(fileSelf, ProjectMapper(Map(project1 -> targetProject1)), targetBase)
+
+  test("Do nothing on a distribution payload without fields to patch") {
+    val input = json"""{ "anotherField": "XXX" }"""
+    patcherNoProjectMapping.single(input).assertEquals(input)
+  }
+
+  test("Patch location on a distribution") {
+    val input    =
+      json"""{
+              "atLocation": {
+                "store": {
+                  "@id": "https://bbp.epfl.ch/remote-disk-storage",
+                  "@type": "RemoteDiskStorage",
+                  "_rev": 3
+                }
+              }
+            }"""
+    val expected =
+      json"""{
+            "atLocation": {
+              "store": {
+                "@id": "https://bluebrain.github.io/nexus/vocabulary/defaultS3Storage",
+                "@type": "S3Storage",
+                "_rev": 1
```

Comment:
It might be worth explaining why these fields are changing, because as a reader I can't tell currently